decider-HSMetrics/src/main/java/ca/on/oicr/pde/deciders/HSMetricsDecider.java

```diff
@@ -0,0 +1,398 @@
+        // Override the supplied group-by value
+        for (ReturnValue currentRV : vals) {
+            boolean metatypeOK = false;
+
+            for (int f = 0; f < currentRV.getFiles().size(); f++) {
+                try {
+                    if (currentRV.getFiles().get(f).getMetaType().equals(BAM_METATYPE)) {
+                        metatypeOK = true;
+                    }
+                } catch (Exception e) {
+                    Log.stderr("Error checking a file");
+                }
+            }
+
+            if (!metatypeOK) {
+                continue; // Go to the next value
+            }
+
+            BeSmall currentSmall = new BeSmall(currentRV);
+            fileSwaToSmall.put(currentRV.getAttribute(groupBy), currentSmall);
+
+            String fileDeets = currentSmall.getIusDetails();
+            Date currentDate = currentSmall.getDate();
+
+            //if there is no entry yet, add it
+            if (iusDeetsToRV.get(fileDeets) == null) {
+                iusDeetsToRV.put(fileDeets, currentRV);
+            } //if there is an entry, compare the current value to the 'old' one in
+            //the map. if the current date is newer than the 'old' date, replace
+            //it in the map
+            else {
+                ReturnValue oldRV = iusDeetsToRV.get(fileDeets);
+                BeSmall oldSmall = fileSwaToSmall.get(oldRV.getAttribute(Header.FILE_SWA.getTitle()));
+                Date oldDate = oldSmall.getDate();
+                if (currentDate.after(oldDate)) {
+                    iusDeetsToRV.put(fileDeets, currentRV);
+                }
+            }
+        }
+
+        //only use those files that entered into the iusDeetsToRV
+        //since it's a map, only the most recent values
+        List<ReturnValue> newValues = new ArrayList<ReturnValue>(iusDeetsToRV.values());
+        Map<String, List<ReturnValue>> map = new HashMap<String, List<ReturnValue>>();
+
+        //group files according to the designated header (e.g. sample SWID)
+        for (ReturnValue r : newValues) {
+            String currVal = fileSwaToSmall.get(r.getAttribute(Header.FILE_SWA.getTitle())).getGroupByAttribute();
+            List<ReturnValue> vs = map.get(currVal);
+            if (vs == null) {
+                vs = new ArrayList<ReturnValue>();
+            }
+            vs.add(r);
+            map.put(currVal, vs);
+        }
+
+        return map;
+    }
+
+    @Override
+    protected String handleGroupByAttribute(String attribute) {
+        String a = super.handleGroupByAttribute(attribute);
+        BeSmall small = fileSwaToSmall.get(a);
+        if (small != null) {
+            return small.getGroupByAttribute();
+        }
+        return attribute;
+    }
+
+    @Override
+    protected Map<String, String> modifyIniFile(String commaSeparatedFilePaths, String commaSeparatedParentAccessions) {
+
+        String[] filePaths = commaSeparatedFilePaths.split(",");
+
+
+
+        for (String p : filePaths) {
```

Comment:
This loop should not encapsulate ini file modification lines. This may become a  problem in case when there are multiple files. I also don't see a check for number of files ensuring that we have only one input.

Reply:
done